### PR TITLE
chores: get flashcards

### PR DESCRIPTION
**Pull request by SzDabrowski**
Fetching data with GET works fine, but i get this strange behaviour - when card gets data, initially it will be empty, but will display text after click event. Also i have to make noneditable card adjust height to given text
image

### CODE REVIEW

**Comment by SzDabrowski** [comment]
At this moment i focused on main funcionality. Should i fix this now or do it later in QA?

File: src/App.tsx [hunk 1]
```diff
@@ -13,52 +13,31 @@ interface FishkappCard {
...
   const addCard = () => {
     setDisplayNewCard(true);
   };
-    
 
   const saveCard = async ( card: FishkappCard) => {
+    setDisplayNewCard(false);
     try {
       const res = await addCardService(card.question, card.answer);
+      await addCardService(card.question, card.answer).then(data => console.log(data));
```
Comment (Sumick) [inline src/App.tsx:38]: **issue:** Looks like `addCardService` is called twice here

**Review by Sumick — CHANGES_REQUESTED** [review]
Looks like posting new card is called twice. You can leave this UI bug for later if its not critical, let's focus on delivering the app

File: src/App.tsx (hunk 1 above)
Comment (SzDabrowski) [reply src/App.tsx]: Issue fixed

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
The goal of the task was to fetch all existing flashcards from the database when initializing the app.

### Requirements
1. Fetch API must be used
2. The following endpoint must be used: `https://training.nerdbord.io/api/v1/fischkapp/flashcards`

### Review

#### src/App.tsx
The changes in this file seem to cover the task requirements. The `getCardsService` function is called inside a `useEffect` hook, which will run when the component mounts, fetching all existing flashcards from the database. The fetched cards are then stored in the `cardsData` state variable. The changes in the properties of the `iCard` interface and the `Card` component props are consistent with the data structure returned by the API.

#### src/components/Card/CardSide.tsx
The changes in this file do not seem to be related to the task requirements. The changes involve adjusting the height of a textarea element, which is not mentioned in the task requirements.

#### src/services/apiService.ts
The `getCardsService` function in this file uses the Fetch API to get data from the specified endpoint, which meets the task requirements. However, there is an unused variable `url` that should be removed or used instead of the hardcoded URL in the `fetch` call. Also, there is an empty `editCardService` function that seems unrelated to the task requirements.

### Conclusion
The code changes mostly cover the task requirements. However, there are some unrelated changes and minor issues that should be addressed.

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task Goal
The task goal was to get all existing flashcards from the database when initializing the app.

### Requirements
1. Fetch API must be used
2. Following endpoint must be used: `https://training.nerdbord.io/api/v1/fischkapp/flashcards`

### Code Review

#### src/App.tsx
The changes in this file seem to cover the task requirements. The `getCardsService` function from `apiService.ts` is imported and used in a `useEffect` hook to fetch the cards from the API when the app initializes. The fetched cards are then stored in the `cardsData` state variable. The structure of the `iCard` interface and the `cardsData` state variable have been updated to match the expected structure of the data returned from the API.

#### src/components/Card/CardSide.tsx
The changes in this file do not seem to be related to the task requirements. The changes involve updating the way the height of a card is set. This does not seem to be related to fetching flashcards from an API.

#### src/services/apiService.ts
The changes in this file cover the task requirements. A new function `getCardsService` has been added which uses the Fetch API to get data from the required endpoint. The function returns a promise that resolves with the fetched data.

### Conclusion
The code changes cover the task requirements. The Fetch API is used to get data from the required endpoint and this data is stored in a state variable when the app initializes. The changes in `CardSide.tsx` do not seem to be related to the task requirements but do not interfere with them either.

**Pull request merged**